cns/restserver/util.go

```diff
@@ -797,10 +829,16 @@ func (service *HTTPRestService) getPrimaryHostInterface(ctx context.Context) (*w
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to get interfaces from IMDS")
 		}
+
 		primary, err := wireserver.GetPrimaryInterfaceFromResult(res)
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to get primary interface from IMDS response")
 		}
+
+		if !isPrivateIP(net.ParseIP(primary.PrimaryIP)) {
+			return nil, errors.Wrap(err, "only private IPv4 or IPv6 address can be used")
+		}
+
 		service.state.primaryInterface = primary
```

Comment:
this is not the interface that metrics get bound to

Reply:
this change is only for standalone scenario.. i guess for aks, cns listening only on localhost